### PR TITLE
9: Register user profile patch endpoint

### PR DESCRIPTION
**Pull request by zainabch123**
Updated updateById function.
Updated API Spec

### CODE REVIEW

**Review by th0jensen — APPROVED** [review]
LGTM

**Review by AtikoSpeed — APPROVED** [review]

File: prisma/schema.prisma [hunk 1]
```diff
@@ -36,7 +36,8 @@ model Profile {
   lastName    String
   bio         String?
   githubUrl   String?
-}
+  profilePicture String?
```
Comment (dogezen) [inline prisma/schema.prisma:39]: you might need an `@default(null)` or something similar - check that you're not getting any warnings in a db with existing prior rows in the profile table

File: src/controllers/user.js [hunk 2]
```diff
@@ -57,11 +57,59 @@ export const getAll = async (req, res) => {
...
+        ...(bio && { bio }),
+        ...(githubUrl && { githubUrl }),
+        ...(profilePicture && { profilePicture })
+      }
+      const updatedUser = await User.updateUser(id, updateData)
+      delete updatedUser.password
+      return sendDataResponse(res, 201, updatedUser)
+    }
+
+    if (req.user.role === 'TEACHER') {
```
Comment (dogezen) [inline src/controllers/user.js:100]: you're saying if I am a teacher, then I can only update the cohortId of a user? what if a user writes offensive text and a teacher wants to update / purge the profile?

I think STUDENT + TEACHER should be able to fully update the profile; except TEACHER can change cohortId on top

File: src/routes/user.js [hunk 3]
```diff
@@ -10,6 +10,6 @@ const router = Router()
 router.post('/', create)
 router.get('/', validateAuthentication, getAll)
 router.get('/:id', validateAuthentication, getById)
-router.patch('/:id', validateAuthentication, validateTeacherRole, updateById)
+router.patch('/:id', validateAuthentication, updateById)
```
Comment (dogezen) [inline src/routes/user.js:13]: good

File: src/controllers/user.js (hunk 2 above)
Comment (dogezen) [reply src/controllers/user.js:100]: Let me know if you decide with this logic anyways :)

**Review by dogezen — CHANGES_REQUESTED** [review]

**Review by dogezen — COMMENTED** [review]
see comments

File: prisma/schema.prisma (hunk 1 above)
Comment (zainabch123) [reply prisma/schema.prisma:39]: profilePicture does not seem to cause errors in db. Previous users already have their profilePicture automatically set to NULL.

File: src/controllers/user.js [hunk 4]
```diff
@@ -57,11 +57,57 @@ export const getAll = async (req, res) => {
...
+  const id = Number(req.params.id)
+  const { firstName, lastName, bio, githubUrl, cohortId, profilePicture } =
+    req.body
 
-  if (!cohortId) {
-    return sendDataResponse(res, 400, { cohort_id: 'Cohort ID is required' })
-  }
+  try {
+    // Check user you want to update exists:
+    const foundUser = await User.findById(id)
```
Comment (dogezen) [inline src/controllers/user.js:66]: i think this should be `await dbClient.User.findById(id)` with `dbClient` imported from utils folder; this to ensure only one prisma connection is being created (as far as I know)

**Review by dogezen — CHANGES_REQUESTED** [review]
minor tweak - also I think eslint/prettier are complaining about quotes and single quotes, maybe run eslint and prettier to double check errors

**Review by dogezen — APPROVED** [review]

**Pull request merged**